process/guardian/guardedAccount.go

```diff
@@ -15,29 +16,34 @@ import (
 
 var guardianKey = []byte(core.ElrondProtectedKeyPrefix + core.GuardiansKeyIdentifier)
 
-const epochsForActivation = 10 // TODO: take from config
-
 type guardedAccount struct {
-	marshaller               marshal.Marshalizer
-	epochNotifier            vmcommon.EpochNotifier
-	mutEpoch                 sync.RWMutex
-	currentEpoch             uint32
-	guardianActivationEpochs uint32
+	marshaller                    marshal.Marshalizer
+	epochNotifier                 vmcommon.EpochNotifier
+	mutEpoch                      sync.RWMutex
+	guardianActivationEpochsDelay uint32
+	currentEpoch                  uint32
 }
 
 // NewGuardedAccount creates a new guarded account
-func NewGuardedAccount(marshaller marshal.Marshalizer, epochNotifier vmcommon.EpochNotifier) (*guardedAccount, error) {
+func NewGuardedAccount(
+	marshaller marshal.Marshalizer,
+	epochNotifier vmcommon.EpochNotifier,
+	setGuardianEpochsDelay uint32,
+) (*guardedAccount, error) {
 	if check.IfNil(marshaller) {
 		return nil, process.ErrNilMarshalizer
 	}
 	if check.IfNil(epochNotifier) {
 		return nil, process.ErrNilEpochNotifier
 	}
+	if setGuardianEpochsDelay == 0 {
```

Comment:
should there be a check with greater than unbondingPeriod? or the fact that someone set it to a value greater than zero should suffice?

Reply:
I have added for now a comment in config, otherwise the system smart contracts config should be passed to the bootstrap component and checked there upon creation.

Reply:
ok